source/enumerate-collections.rst

```diff
@@ -265,7 +266,17 @@ Example return::
         "system.replset"
     ]
 
-Drivers MUST strip the database name from the returned collection names.
+
+
+Drivers MAY implement a MongoClient method that returns an Iterable of strings,
+each of which corresponds to a name in the collections array of the
+``listCollections`` command result. This method SHOULD be named
+``listCollections``.
+
+Older versions of the server that do not support the ``nameOnly`` option for the
```

Comment:
If we move the "MongoDB 4.0 introduces..." paragraph to prefix this text, you can then say that "versions of the server between 2.7.6 (inclusive) and 4.0 (exclusive) do not support the `nameOnly` option...". This will purposefully omit earlier server versions, which rely on queries to `system.namespaces`.